### PR TITLE
Add configurable ping sounds

### PR DESCRIPTION
**Pull request by zhliau**
Adds a configuration menu option to allow the user to set the ping sound that plays on in-game ping receive/send. I thought the Subquest complete sound was a bit overbearing.

Feedback is much appreciated!

### CODE REVIEW

File: Fika.Core/Coop/Factories/PingFactory.cs [hunk 1]
```diff
@@ -22,6 +23,24 @@ public enum EPingType : byte
         Interactable
     }
 
+    public static EUISoundType GetPingSound()
+    {
+        return FikaPlugin.PingSound.Value switch
```
Comment (zhliau) [inline Fika.Core/Coop/Factories/PingFactory.cs:28]: I suppose I could use a map of some sort for this, but I don't know C# well enough to be confident it's the right approach and this is the first Unity anything I've done.

File: Fika.Core/FikaPlugin.cs [hunk 2]
```diff
@@ -136,6 +137,7 @@ public class FikaPlugin : BaseUnityPlugin
         public static ConfigEntry<bool> PingUseOpticZoom { get; set; }
         public static ConfigEntry<bool> PingScaleWithDistance { get; set; }
         public static ConfigEntry<float> PingMinimumOpacity { get; set; }
+        public static ConfigEntry<EPingSound> PingSound { get; set; }
```
Comment (Lacyway) [inline Fika.Core/FikaPlugin.cs:140]:
```suggestion
        public static ConfigEntry<EUISoundType> PingSound { get; set; }
```
Could you not use the enum here, this way you don't need the helper function?

**Review by Lacyway — CHANGES_REQUESTED** [review]

File: Fika.Core/FikaPlugin.cs (hunk 2 above)
Comment (zhliau) [reply Fika.Core/FikaPlugin.cs:140]: That definitely works, but I didn't want to flood the config menu with a bunch of sounds like `PlayerIsDead` or `GeneratorTurnOn` which play a bit too long to be appropriate for a 'pingy' sound. I can definitely change it if you feel it's better to provide all options though.

Comment (Lacyway) [reply Fika.Core/FikaPlugin.cs:140]: > That definitely works, but I didn't want to flood the config menu with a bunch of sounds like `PlayerIsDead` or `GeneratorTurnOn` which play a bit too long to be appropriate for a 'pingy' sound. I can definitely change it if you feel it's better to provide all options though.
> 
> Is there a way to limit the options for EUISoundType?

I never opened the editor and forgot that these existed. Good point!

**Review by Lacyway — APPROVED** [review]

**Comment by Lacyway** [comment]
@zhliau please DM me on the discord.

**Pull request merged**